sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1092,13 +1082,30 @@ object CodeGenerator extends Logging {
         logInfo(s"\n${CodeFormatter.format(code, maxLines)}")
         throw new CompileException(msg, e.getLocation)
     }
+
+    // Check if compiled code has a too large function
+    methodsToByteCodeSize.foreach { case (name, byteCodeSize) =>
+      if (byteCodeSize > SQLConf.get.hugeMethodLimit) {
+        val clazzName = evaluator.getClazz.getSimpleName
+        val methodName = name.replace("$", "")
+        val msg = s"failed to compile: the size of $clazzName.$methodName was $byteCodeSize and " +
+          "this value went over the limit `spark.sql.codegen.hugeMethodLimit`" +
+          s"(${SQLConf.get.hugeMethodLimit}). To avoid this error, you can make this limit higher."
+        logError(msg)
+        val maxLines = SQLConf.get.loggingMaxLinesForCodegen
+        logInfo(s"\n${CodeFormatter.format(code, maxLines)}")
```

Comment:
Is it useful to log the gen'd code for this case? The code gets compiled so it might not helpful to print out the code.

Reply:
aha, I see. I also think this info is less meaning for users, but some meaningful for developers? So, how about changing to `logDebug`?

Reply:
We already have `logDebug` at line 1072. We don't need to log it again.

Reply:
aha, ok. I'll drop.